tasks/eva_2406/gather_release_species.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python
+import csv
+import json
+import operator
+import os
+import re
+from argparse import ArgumentParser
+from collections import defaultdict
+
+import atexit
+
+import pandas as pd
+import psycopg2
+import requests
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.pg_utils import execute_query, get_all_results_for_query
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+
+eutils_url = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/'
+esearch_url = eutils_url + 'esearch.fcgi'
+esummary_url = eutils_url + 'esummary.fcgi'
+efetch_url = eutils_url + 'efetch.fcgi'
+ensembl_url = 'http://rest.ensembl.org/info/assembly'
+
+
+cache_file = 'cache.json'
+
+
+def load_cache():
+    global cache
+    if os.path.exists(cache_file):
+        with open(cache_file) as open_file:
+            cache.update(json.load(open_file))
+
+
+def save_cache():
+    with open(cache_file, 'w') as open_file:
+        json.dump(cache, open_file)
+
+
+atexit.register(save_cache)
+
+
+cache = defaultdict(dict)
+load_cache()
+
+
+def retrieve_assembly_summary_from_species_name(species):
```

Comment:
This method and the next two seem unused right now, maybe you're keeping them around for the future though...

Reply:
That is correct, they might be used at a later date. They are meant to retrieve the most recent or most complete assembly when there isn't an official Ensembl assembly available. But for now I have not used them because I want to discuss this with Ensembl first.